src/Jackett.Common/Indexers/IPTorrents.cs

```diff
@@ -228,48 +231,45 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                 "western"
             };
 
-
-            /* notes: 
+            /* notes:
              * IPTorrents can search for genre (tags) using the default title&tags search
-             * qf= 
+             * qf=
              * "" = Title and Tags
              * ti = Title
              * ta = Tags
              * all = Title, Tags & Description
              * adv = Advanced
-             * 
+             *
              * But only movies and tv have tags.
              */
 
+            var headers = new Dictionary<string, string>();
+            if (!string.IsNullOrEmpty(configData.UserAgent.Value))
+                headers.Add("User-Agent", configData.UserAgent.Value);
+
             var qc = new NameValueCollection();
 
+            foreach (var cat in MapTorznabCapsToTrackers(query))
+                qc.Set(cat, string.Empty);
 
-            Dictionary<string, string> headers = null;
+            if (((BoolConfigurationItem)configData.GetDynamic("freeleech")).Value)
+                qc.Set("free", "on");
 
-            if (!string.IsNullOrEmpty(configData.UserAgent.Value))
-            {
-                headers = new Dictionary<string, string>
-                {
-                    { "User-Agent", configData.UserAgent.Value }
-                };
-            }
+            var searchQuery = new List<string>();
 
+            // IPT uses sphinx, which supports boolean operators and grouping
             if (query.IsImdbQuery)
-                qc.Add("q", query.ImdbID);
-            else
-            if (query.IsGenreQuery)
-                qc.Add("q", query.GetQueryString() + " " + query.Genre);
-            else
+                searchQuery.Add($"+({query.ImdbID})");
+            else if (query.IsGenreQuery)
```

Comment:
Does this still need to be `else if`?

Reply:
Searching `tt13146488` should only return results for Peacemaker, but for whatever reason it's also returning some Flash results too. In that case, the genres are the same, so it wouldn't help to search by them, but if they weren't then it'd be good to have the option search by imdbid and genre.

Reply:
Had a quick look at other indexers which support both and there's a mixture of `if` and `else if`. Your point makes sense, so I'm happy to merge as is.